### PR TITLE
[MoistThermo] use PhasePartition object

### PR DESCRIPTION
**Pull request by simonbyrne**
As discussed in #147.

### CODE REVIEW

File: docs/src/Utilities/MoistThermodynamics.md [hunk 1]
```diff
@@ -28,47 +28,46 @@ There are several types of functions:
...
+geopotential = grav * z
+q_tot          = ...
+ρ            = ...
 
-    # initialize
-    geopotential = grav * z
-    q_tot          = ...
-    ρ            = ...
+(u, v, w)    = ...
+e_kin           = 0.5 * (u.^2 .+ v.^2 .+ w.^2)
```
Comment (charleskawczynski) [inline docs/src/Utilities/MoistThermodynamics.md]:
```suggestion
e_kin           = 0.5 * (u^2 + v^2 + w^2)
```
I think this is typically done element-wise, should we remove the `.`?

File: docs/src/Utilities/MoistThermodynamics.md [hunk 2]
```diff
@@ -28,47 +28,46 @@ There are several types of functions:
...
-    do timestep   # timestepping loop
+  # advance dynamical variables by a timestep (temperature typically
+  # appears in terms on the rhs, such as radiative transfer)
+  advance(u, v, w, ρ, e_tot, q_tot)
 
-      # advance dynamical variables by a timestep (temperature typically
-      # appears in terms on the rhs, such as radiative transfer)
-      advance(u, v, w, ρ, e_tot, q_tot)
+  # compute internal energy from dynamic variables
+  e_int = e_tot - 0.5 * (u.^2 .+ v.^2 .+ w.^2) - geopotential
```
Comment (charleskawczynski) [inline docs/src/Utilities/MoistThermodynamics.md]:
```suggestion
  e_int = e_tot - 0.5 * (u^2 + v^2 + w^2) - geopotential
```
I think this is typically done element-wise, should we remove the `.`?

**Comment by simonbyrne** [comment]
> Still looking through this, but I think this makes sense. It protects against passing in `q_tot`, `q_liq` and `q_ice` in the wrong order, which is really nice. My only immediate suggestion might be to change the name of the `q` constructor to avoid TitleCase and `_`'s:
> 
> `PhasePartition_equil` -> `PhasePartitionEquil`

Yeah, it's not ideal, I agree. I'm not that keen `PhasePartitionEquil` as it doesn't return an object of that name.

**Comment by charleskawczynski** [comment]
> I'm not that keen PhasePartitionEquil as it doesn't return an object of that name.

True. I suppose that you could make it an outer constructor of `PhasePartition`, and use keyword args, but then the user needs to remember the default behavior. I'm fine with whatever you prefer.

**Pull request merged**